app/src/main/java/org/mozilla/focus/settings/ManualAddSearchEngineSettingsFragment.kt

```diff
@@ -206,7 +206,11 @@ class ManualAddSearchEngineSettingsFragment : BaseSettingsFragment() {
             )
 
             return try {
-                client.fetch(request).status < VALID_RESPONSE_CODE_UPPER_BOUND
+                val response = client.fetch(request)
+                // Close the response stream to ensure the body is closed correctly. See https://bugzilla.mozilla.org/show_bug.cgi?id=1603114.
+                response.body.close()
```

Comment:
Why does this not happen in Fenix as well? 🤔

Reply:
It did. In Fenix, we use `response.body.string()` to read the stream (that also automatically closes it. ) At the moment of the fix the `close()` method had a bug that has been fixed now.  We should also use close there too. I will put up a PR for that.

Reply:
https://github.com/mozilla-mobile/fenix/pull/21203